tests/test_taskgroups.py

```diff
@@ -542,6 +542,23 @@ async def cancel_when_ready():
     assert not sleep_completed
 
 
+async def test_shielding_mutate():
+
+    async def task(task_status):
+        with CancelScope() as scope:
+            await sleep(.1)
```

Comment:
the intent of the sleep is to have a checkpoint before the shield False -> True transition.  Otherwise the intended distinction with CancelScope(shield=True) is diminished.

Reply:
Not really – it does test that the setter works as intended when the flag is flipped. Since the scope is only cancelled after `task_status.started()`, the checkpoint doesn't do anything useful.

Reply:
I agree that the checkpoint doesn't do anything useful given our knowledge of the current implementation.  But if I were to write a test to validate a range of known and unknown implementations, then enabling the shield some time after the cancel scope is entered seems more likely to shake out bugs.